### DISCUSSION TITLE
My USB-MIDI cable stopped working under Linux (but seems to work under Windows)

### DISCUSSION THREAD

**Discussion by kerkovits** [GrandOrgue/grandorgue#1825] [Help] [answered]
I know that such issues are hard to investigate, but maybe someone has more knowledge about the deep emotions of a Linux system.

My organ console has a MIDI output, and it is connected to my PC via an USB-to-MIDI adapter. It stopped working after a system upgrade.

It does not seem to be a hardware fault: exactly the same cable on exactly the same port works fine if I boot the same computer to Windows (it is a dual-boot computer). Under Linux, I get the following logs repeatedly each second:
```
$ dmesg -w
[ 1112.366637] usb 1-11: new full-speed USB device number 34 using xhci_hcd
[ 1112.494359] usb 1-11: New USB device found, idVendor=fc02, idProduct=0101, bcdDevice= 2.50
[ 1112.494373] usb 1-11: New USB device strings: Mfr=0, Product=2, SerialNumber=0
[ 1112.494379] usb 1-11: Product: USB MIDI Interface
[ 1112.496547] usb 1-11: Quirk or no altest; falling back to MIDI 1.0
[ 1112.504466] usb 1-11: USB disconnect, device number 34
[ 1112.918631] usb 1-11: new full-speed USB device number 35 using xhci_hcd
[ 1113.046671] usb 1-11: New USB device found, idVendor=fc02, idProduct=0101, bcdDevice= 2.50
[ 1113.046675] usb 1-11: New USB device strings: Mfr=0, Product=2, SerialNumber=0
[ 1113.046676] usb 1-11: Product: USB MIDI Interface
[ 1113.048220] usb 1-11: Quirk or no altest; falling back to MIDI 1.0
[ 1113.052474] usb 1-11: USB disconnect, device number 35
```
Of course, `lsusb` does not list the device. Multiple USB ports were tried.

I tried downgrading packages kernel, alsa, and linux-firmware without success.

Before buying a new USB-MIDI adapter, I have to confirm that it is a hardware fault, but as it works under Windows without problems, I am afraid that the error is not in the cable, so buying a new one would be useless.

**Comment by oleg68** [comment]
> It stopped working after a system upgrade.

It looks like a linux kernel issue. Could you try to load with the old kernel?

**Reply by kerkovits** [reply]
Thank you very much for your help. It was also my first thought, but it did not change the situation.

**Reply by oleg68** [reply]
What kind of system upgrade had you performed before it stopped working?

**Reply by kerkovits** [reply]
Just a simple `sudo dnf up`. It updated a plenty of shared libraries and linux-firmware. It also installed a new kernel, but I can boot into the older one.

**Reply by oleg68** [reply]
Do you have a backup of the old system?

**Reply by kerkovits** [reply]
No.

**Reply by oleg68** [reply]
Then try to do full power off your computer (with unplugging from the socket) and wait couple of minutes. Sometimes usb devices star working after cold reset.

**Reply by kerkovits** [reply]
I plug the PC out every night, so I have already done it.

**Comment by oleg68** [comment]
@kerkovits I experienced the same problem with my midi-usb cable after I updated my Fedora 39 system.

But I performed a backup before ``dnf update``.

Now I have restored the old f39 from the backup and everething seems to work. This evening I'm going to update packages one-by-one for determining, which one causes the problem.

**Comment by oleg68** [comment, marked as answer]
@kerkovits I found that `fwupd-1.9.14`  breaks midi-to-usb cable to work.

After downgrading it to fwupd-1.9.5 everything works as expected.

So if you have made ``dnf update``, you have to perform ``dnf downgrade 'fwupd*'`` fore restoring your midi-to-usb to work.

**Reply by kerkovits** [reply]
Thank you very much for your time, it worked.

**Reply by kerkovits** [reply]
In fact, a new version of fwupd arrived to fedora repos today. The cable works also with that one.

**Reply by oleg68** [reply]
Seems there were bugs https://github.com/fwupd/fwupd/issues/6882 and https://github.com/fwupd/fwupd/issues/6871

They were fixed in 1.9.15

**Reply by kerkovits** [reply]
It is good to know that it is not a hardware fault. I had two BSOD on Windows last week while using GO, which also made me suspect that the cable is not working perfectly.
